gateway/src/apicast/policy/conditional/condition.lua

```diff
@@ -0,0 +1,54 @@
+local setmetatable = setmetatable
+local ipairs = ipairs
+local assert = assert
+
+local _M = {}
+
+local mt = { __index = _M }
+
+local default_combine_op = 'and'
+
+local supported_combine_ops = { ["and"] = 1 , ["or"] = 1 }
+
+local function all_true(operations, context)
+  for _, operation in ipairs(operations) do
+    if not operation:evaluate(context) then
+      return false
+    end
+  end
+
+  return true
+end
+
+local function at_least_one_true(operations, context)
+  for _, operation in ipairs(operations) do
+    if operation:evaluate(context) then
+      return true
+    end
+  end
+
+  return false
+end
+
+function _M.new(operations, combine_op)
+  local self = setmetatable({}, mt)
+
+  self.operations = operations or {}
+  self.combine_op = combine_op or default_combine_op
+
+  assert(supported_combine_ops[self.combine_op], 'Unsupported combine op')
+
+  return self
+end
+
+function _M:evaluate(context)
+  if #self.operations == 0 then return true end
+
+  if self.combine_op == 'and' then
```

Comment:
This can be done at initialization time, right? 

It would allow for better JIT if there would be no branching in this method.

Reply:
Right :+1: 
Fixed.